autofit/plot/samples_text.py

```diff
@@ -0,0 +1,150 @@
+import configparser
+import logging
+
+from autofit import conf
+from autofit.tools import text_util, text_formatter
+
+logger = logging.getLogger(__name__)
+
+def results_from_sigma(samples, sigma) -> str:
+    """ Create a string summarizing the results of the non-linear search at an input sigma value.
+
+    This function is used for creating the model.results files of a non-linear search.
+
+    Parameters
+    ----------
+    sigma : float
+        The sigma within which the PDF is used to estimate errors (e.g. sigma = 1.0 uses 0.6826 of the PDF)."""
+    try:
+        lower_limits = samples.vector_at_lower_sigma(sigma=sigma)
+        upper_limits = samples.vector_at_upper_sigma(sigma=sigma)
+    except ValueError:
+        return ""
+
+    sigma_formatter = text_formatter.TextFormatter()
+
+    for i, prior_path in enumerate(samples.model.unique_prior_paths):
+        value = samples.format_str.format(samples.most_probable_vector[i])
+        upper_limit = samples.format_str.format(upper_limits[i])
+        lower_limit = samples.format_str.format(lower_limits[i])
+        value = value + " (" + lower_limit + ", " + upper_limit + ")"
+        sigma_formatter.add((prior_path, value))
+
+    return "\n\nMost probable model ({} sigma limits):\n\n{}".format(
+        sigma, sigma_formatter.text
+    )
+
+
+def latex_results_at_sigma(samples, sigma, format_str="{:.2f}") -> [str]:
+    """Return the results of the non-linear search at an input sigma value as a string that is formated for simple
+    copy and pasting in a LaTex document.
+
+    Parameters
+    ----------
+    sigma : float
+        The sigma within which the PDF is used to estimate errors (e.g. sigma = 1.0 uses 0.6826 of the PDF).
+    format_str : str
+        The formatting of the parameter string, e.g. how many decimal points to which the parameter is written.
+    """
+
+    labels = samples.param_labels
+    most_probables = samples.most_probable_vector
+    uppers = samples.vector_at_upper_sigma(sigma=sigma)
+    lowers = samples.vector_at_lower_sigma(sigma=sigma)
+
+    line = []
+
+    for i in range(len(labels)):
+        most_probable = format_str.format(most_probables[i])
+        upper = format_str.format(uppers[i])
+        lower = format_str.format(lowers[i])
+
+        line += [
+            labels[i]
```

Comment:
Definitely a case for an fstring

You would need to escape the curly braces with \ I think

Reply:
Couldnt get it to work.